Lesson8/src/by/home/service/FileUtil.java

```diff
@@ -0,0 +1,149 @@
+    private final String newDirName;
+    private final String zipName;
+    private final String FILE_NAME_1 = "bio.txt";
+    private final String FILE_NAME_2 = "info.txt";
+    private final String FILE_NAME_3 = "demo.txt";
+
+    public FileUtil(String dirPath, String dirName, String zipName, String newDirName) {
+        this.dirPath = dirPath;
+        this.dirName = dirName;
+        this.zipName = zipName;
+        this.newDirName = newDirName;
+    }
+
+
+    public void workWithFiles() throws IOException {
+        File dir = createDir();
+
+        File file1 = createFile(dir, FILE_NAME_1);
+        File file2 = createFile(dir, FILE_NAME_2);
+        File file3 = createFile(dir, FILE_NAME_3);
+
+        writeInfoToFile(file1);
+        writeInfoToFile(file2);
+        writeInfoToFile(file3);
+
+        List<File> fileList = new LinkedList<>();
+        fileList.add(file1);
+        fileList.add(file2);
+        fileList.add(file3);
+
+        createZip(zipName, fileList);
+
+        File newDir = reNameDir(dir);
+
+        printAllIncludes(newDir);
+
+        deleteDir(newDir);
+    }
+
+    private File createDir() {
+        File dir = new File(dirPath + "/" + dirName);
+        boolean isExists = dir.exists();
+        boolean created = dir.mkdir();
+        if (isExists) {
+            System.out.println("Directory is already exists!");
+            return dir;
+        } else if (created) {
+            System.out.println("Directory was successfully created!");
+            return dir;
+        } else {
+            System.out.println("Directory was not created!");
+            return null;
+        }
+    }
+
+    private void printAllIncludes(File dir) {
+        if (dir.isDirectory()) {
+            System.out.println("The list of files in dir: ");
+            for (File item : dir.listFiles()) {
+                System.out.println(item.getName());
+            }
+        }
+    }
+
+    private File reNameDir(File dir) {
+        File newDir = new File(dirPath + "/" + newDirName);
+        if (dir.renameTo(newDir)) {
+            System.out.println("Dir renamed!");
+        }
+        return newDir;
+    }
+
+    private void deleteDir(File file) throws FileNotFoundException {
+        if (file.isDirectory() && file.listFiles().length != 0) {
+            for (File item : file.listFiles()) {
+                deleteFile(item);
+            }
+        }
+        if (file.listFiles().length == 0) {
+            deleteFile(file);
+        }
+
+    }
+
+    private File createFile(File dir, String fileName) throws IOException {
+        File file = new File(dir, fileName);
+        if (file.exists()) {
+            System.out.println("File " + fileName + " exists!");
+            return file;
+        } else if (file.createNewFile()) {
+            System.out.println("File " + fileName + " has been created!");
+            return file;
+        } else {
+            System.out.println("File " + fileName + " not found!");
```

Comment:
I think it is more about the situation when you were trying to create file, but it didn't happen. So I would say that the message "Unable to create file " + fileName would make more sense to me